vendor/raygui/raygui.odin

```diff
@@ -0,0 +1,555 @@
+    SLIDER_PADDING,             // Slider/SliderBar internal bar padding
+}
+
+// ProgressBar
+GuiProgressBarProperty :: enum c.int {
+    PROGRESS_PADDING = 16,      // ProgressBar internal padding
+}
+
+// ScrollBar
+GuiScrollBarProperty :: enum c.int {
+    ARROWS_SIZE = 16,
+    ARROWS_VISIBLE,
+    SCROLL_SLIDER_PADDING,      // (SLIDERBAR, SLIDER_PADDING)
+    SCROLL_SLIDER_SIZE,
+    SCROLL_PADDING,
+    SCROLL_SPEED,
+}
+
+// CheckBox
+GuiCheckBoxProperty :: enum c.int {
+    CHECK_PADDING = 16,          // CheckBox internal check padding
+}
+
+// ComboBox
+GuiComboBoxProperty :: enum c.int {
+    COMBO_BUTTON_WIDTH = 16,    // ComboBox right button width
+    COMBO_BUTTON_SPACING,       // ComboBox button separation
+}
+
+// DropdownBox
+GuiDropdownBoxProperty :: enum c.int {
+    ARROW_PADDING = 16,         // DropdownBox arrow separation from border and items
+    DROPDOWN_ITEMS_SPACING,     // DropdownBox items separation
+}
+
+// TextBox/TextBoxMulti/ValueBox/Spinner
+GuiTextBoxProperty :: enum c.int {
+    TEXT_INNER_PADDING = 16,    // TextBox/TextBoxMulti/ValueBox/Spinner inner text padding
+    TEXT_LINES_SPACING,         // TextBoxMulti lines separation
+    TEXT_ALIGNMENT_VERTICAL,    // TextBoxMulti vertical alignment: 0-CENTERED, 1-UP, 2-DOWN
+    TEXT_MULTILINE,             // TextBox supports multiple lines
+    TEXT_WRAP_MODE,             // TextBox wrap mode for multiline: 0-NO_WRAP, 1-CHAR_WRAP, 2-WORD_WRAP
+}
+
+// Spinner
+GuiSpinnerProperty :: enum c.int {
+    SPIN_BUTTON_WIDTH = 16,     // Spinner left/right buttons width
+    SPIN_BUTTON_SPACING,        // Spinner buttons separation
+}
+
+// ListView
+GuiListViewProperty :: enum c.int {
+    LIST_ITEMS_HEIGHT = 16,     // ListView items height
+    LIST_ITEMS_SPACING,         // ListView items separation
+    SCROLLBAR_WIDTH,            // ListView scrollbar size (usually width)
+    SCROLLBAR_SIDE,             // ListView scrollbar side (0-left, 1-right)
+}
+
+// ColorPicker
+GuiColorPickerProperty :: enum c.int {
+    COLOR_SELECTOR_SIZE = 16,
+    HUEBAR_WIDTH,               // ColorPicker right hue bar width
+    HUEBAR_PADDING,             // ColorPicker right hue bar separation from panel
+    HUEBAR_SELECTOR_HEIGHT,     // ColorPicker right hue bar selector height
+    HUEBAR_SELECTOR_OVERFLOW,   // ColorPicker right hue bar selector overflow
+}
+
+SCROLLBAR_LEFT_SIDE :: 0
+SCROLLBAR_RIGHT_SIDE :: 1
+
+//----------------------------------------------------------------------------------
+// Global Variables Definition
+//----------------------------------------------------------------------------------
+// ...
+
+//----------------------------------------------------------------------------------
+// Module Functions Declaration
+//----------------------------------------------------------------------------------
+
+@(default_calling_convention="c")
+foreign lib {
+    // Global gui state control functions
```

Comment:
Leave an extra blank line below comments like this, as not to associate them with the declaration below---this is how the automatic documentation tool works.

Reply:
Done, please verify.